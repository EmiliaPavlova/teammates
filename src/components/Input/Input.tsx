type Props = {
  id: string;
  label: string;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  value?: string;
  placeholder?: string;
  error?: string;
}

export default function Input({ id, label, value, placeholder, onChange, error }: Props) {
  return (
    <label className="form-item" htmlFor={id}>
      <span className="form-item-label">{label}</span>
      <input
        id={id}
        value={value}
        placeholder={placeholder}
        onChange={onChange}
      />
      {error
        ? <span className="form-item-error">{error}</span>
        : null}
    </label>
  )
}