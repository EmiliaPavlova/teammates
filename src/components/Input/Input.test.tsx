import { shallow } from 'enzyme';
import Input from './Input';

describe('Input component', () => {
  it('renders the component with all props', () => {
    const wrapper = shallow(<Input
      id='Name'
      label='Name'
      value='John Doe'
      placeholder='enter name'
      onChange={jest.fn()}
      error='error'
    />);
    expect(wrapper).toMatchSnapshot();
  })

  it('renders the component without an error', () => {
    const wrapper = shallow(<Input
      id='Name'
      label='Name'
      value='John Doe'
      placeholder='enter name'
      onChange={jest.fn()}
    />);
    expect(wrapper).toMatchSnapshot();
  })
})