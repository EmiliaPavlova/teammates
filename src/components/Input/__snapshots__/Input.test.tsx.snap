// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component renders the component with all props 1`] = `
<label
  className="form-item"
  htmlFor="Name"
>
  <span
    className="form-item-label"
  >
    Name
  </span>
  <input
    id="Name"
    onChange={[MockFunction]}
    placeholder="enter name"
    value="John Doe"
  />
  <span
    className="form-item-error"
  >
    error
  </span>
</label>
`;

exports[`Input component renders the component without an error 1`] = `
<label
  className="form-item"
  htmlFor="Name"
>
  <span
    className="form-item-label"
  >
    Name
  </span>
  <input
    id="Name"
    onChange={[MockFunction]}
    placeholder="enter name"
    value="John Doe"
  />
</label>
`;
